packages/cardano-services/src/Asset/CardanoTokenRegistry.ts

```diff
@@ -0,0 +1,142 @@
+import { Asset, Cardano } from '@cardano-sdk/core';
+import { InMemoryCache } from '../InMemoryCache';
+import { Logger } from 'ts-log';
+import { Shutdown } from '@cardano-sdk/util';
+import { TokenMetadataService } from './types';
+import axios, { AxiosInstance } from 'axios';
+
+const DEFAULT_METADATA_SERVER_URI = 'https://tokens.cardano.org';
+
+/**
+ * Configuration options for CardanoTokenRegistry
+ */
+export interface CardanoTokenRegistryConfiguration {
+  /**
+   * The cache TTL in seconds. Default: 10 minutes.
+   */
+  cacheTTL?: number;
+
+  /**
+   * The Cardano Token Registry public API base URL
+   */
+  metadataServerUri?: string;
+}
+
+interface CardanoTokenRegistryConfigurationWithRequired extends CardanoTokenRegistryConfiguration {
+  cacheTTL: number;
+  metadataServerUri: string;
+}
+
+/**
+ * Dependencies that are need to create CardanoTokenRegistry
+ */
+export interface CardanoTokenRegistryDependencies {
+  /**
+   * The cache engine. Default: InMemoryCache with CardanoTokenRegistryConfiguration.cacheTTL as default TTL
+   */
+  cache?: InMemoryCache;
+
+  /**
+   * The logger object
+   */
+  logger: Logger;
+}
+
+/**
+ * TokenMetadataService implementation using Cardano Token Registry public API
+ */
+export class CardanoTokenRegistry implements TokenMetadataService, Shutdown {
+  /**
+   * The axios client used to retrieve metadata from API
+   */
+  #axiosClient: AxiosInstance;
+
+  /**
+   * The in memory cache engine
+   */
+  #cache: InMemoryCache;
+
+  /**
+   * The logger object
+   */
+  #logger: Logger;
+
+  constructor({ cache, logger }: CardanoTokenRegistryDependencies, config: CardanoTokenRegistryConfiguration = {}) {
+    const defaultConfig: CardanoTokenRegistryConfigurationWithRequired = {
+      cacheTTL: 600,
+      metadataServerUri: DEFAULT_METADATA_SERVER_URI,
+      ...config
+    };
+
+    this.#cache = cache || new InMemoryCache(defaultConfig.cacheTTL);
+    this.#axiosClient = axios.create({ baseURL: defaultConfig.metadataServerUri });
+    this.#logger = logger;
+  }
+
+  shutdown() {
```

Comment:
While working on ADP-1606 I noticed that providers should expose close `method` rather than a `shutdown` one ([ref here](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/cardano-services/src/Http/HttpService.ts#L46)). How do you think it is better to proceed?

Reply:
We should unify the terminology, but let's not worry about it in this ticket.